### PR TITLE
Add runtimeconfig to dskit

### PR DESCRIPTION
**Pull request by treid314**

### CODE REVIEW

File: runtimeconfig/manager.go [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+
+	configMtx sync.RWMutex
+	config    interface{}
+
+	configLoadSuccess prometheus.Gauge
+	configHash        *prometheus.GaugeVec
+}
+
+// New creates an instance of Manager and starts reload config loop based on config
+func New(cfg Config, registerer prometheus.Registerer, logger log.Logger) (*Manager, error) {
```
Comment (treid314) [inline runtimeconfig/manager.go:61]: Should we add a prefix here for the metric names?

File: runtimeconfig/manager.go [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+type Config struct {
+	ReloadPeriod time.Duration `yaml:"period"`
+	// LoadPath contains the path to the runtime config file, requires an
+	// non-empty value
+	LoadPath string `yaml:"file"`
+	Loader   Loader `yaml:"-"`
+}
+
+// RegisterFlags registers flags.
+func (mc *Config) RegisterFlags(f *flag.FlagSet) {
```
Comment (treid314) [inline runtimeconfig/manager.go:37]: We could add a prefix here, but it would a be a breaking change for cortex/tempo/loki

Comment (pstibrany) [reply runtimeconfig/manager.go:37]: I would keep it as is for now. We can always add `RegisterFlagsWithPrefix` in the future, if needed.

File: runtimeconfig/manager.go (hunk 1 above)
Comment (pstibrany) [reply runtimeconfig/manager.go:61]: Generic is fine. People can pass custom registrerer that adds prefix, if they want to. (`prometheus.WrapRegistererWithPrefix`)

**Review by pstibrany — APPROVED** [review]
LGTM, thank you.

**Review by pracucci — APPROVED** [review]
Good job!

**Pull request merged**